### PR TITLE
bump version to 1.4.0 and update changelog

### PR DESCRIPTION
**Pull request by mattsb42-aws**
*Description of changes:*
Prepping for a `1.4.0` release today.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,6 +11,15 @@ Minor
 * Remove dependence on all ``source_stream`` APIs except for ``read()``.
   `#103 <https://github.com/aws/aws-encryption-sdk-python/issues/103>`_
 
+Potentially Breaking
+^^^^^^^^^^^^^^^^^^^^
+
+* Encryption streams no longer close the ``source_stream`` when they themselves close.
+  If you are using context managers for all of your stream handling,
+  this change will not affect you.
+  However, if you have been relying on this to close your ``source_stream`` for you,
```
Comment (lizroth) [inline CHANGELOG.rst]: Would you mind expanding "this" into a class name or at least a more specific reference, to try to be as pedantically clear as possible to someone reading this quickly?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,6 +11,15 @@ Minor
 * Remove dependence on all ``source_stream`` APIs except for ``read()``.
   `#103 <https://github.com/aws/aws-encryption-sdk-python/issues/103>`_
 
+Potentially Breaking
```
Comment (lizroth) [inline CHANGELOG.rst]: Minor quibble: I might call this "potentially backwards incompatible" instead, but not a strong preference.

Comment (mattsb42-aws) [reply CHANGELOG.rst]: Sure.

**Comment by mattsb42-aws** [comment]
Thanks, past me, for making sure that we actually removed deprecated things when they need to go. :)

File: CHANGELOG.rst (hunk 1 above)
Comment (mattsb42-aws) [reply CHANGELOG.rst]: done

**Pull request merged**